src/otk/traversal.py

```diff
@@ -16,5 +19,7 @@ def copy(self, *, path=None, defines=None) -> "State":
             path = self.path
         if defines is None:
             defines = self.defines
+        if includes is None:
+            includes = self.includes.copy()
```

Comment:
Late to the party but:
Should we make `copy()` "smarter" and add something like:
```
if path != includes[-1]:
  includes.append(path)
```
so that whenever the path changes it's automatically added to the include stack?

Too smart?